server/clip_server/executors/clip_torch.py

```diff
@@ -72,33 +73,43 @@ def __init__(
             # For more details, please see https://pytorch.org/docs/stable/generated/torch.set_num_threads.html
             torch.set_num_threads(max(num_threads, 1))
             torch.set_num_interop_threads(1)
+
+        self._num_worker_preprocess = num_worker_preprocess
         self._pool = ThreadPool(processes=num_worker_preprocess)
 
         self._model = CLIPModel(name, device=self._device, jit=jit, **kwargs)
         self._tokenizer = Tokenizer(name)
         self._image_transform = clip._transform_ndarray(self._model.image_size)
 
+        if not self.tracer:
+            self.tracer = NoOpTracer()
```

Comment:
It seems that the tracker is enabled by default. Is it possible to use it as an optional function?

Reply:
The `NoOpTracer` is a no operations implementation means that it's a pass through, no overhead implementation that is available in the jina[standard] installation. This is a requirement to have a no-operations tracer otherwise using the context managers like `with self.tracer.start_as_current_span` will become more complicated. The BaseExecutor already sets the value to None but the NoOpTracer should be used for custom spans even if tracing is turned off.